### PR TITLE
created provider model and added create provider function

### LINKED ISSUE #79: Epic 2: Becoming a Provider: Create a Provider
Technical Tasks:

Create Sign Up Form
Send the form details to the backend
Create Post in backend
Insert a Provider in the Database
Create Provider Table in Database

Show message that provider was created in Front end

### PR DESCRIPTION
**Pull request by karleenmsrichards**
## Description

created Provider Model and function to create a Provider.

## Related to
#79 

Fixes #79 

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have carefully reviewed my own code
- [ ] I have commented my code
- [ ] I have updated any documentation

### CODE REVIEW

File: server/controller/apiController.js [hunk 1]
```diff
@@ -1,6 +1,53 @@
...
+			country,
+			profession,
+			yearsOfExperience,
+			hourlyRate,
+			language,
+		});
+
+		return { message: "Provider created successfully", provider: newProvider };
+	} catch (error) {
+		return { message: "Error creating Provider" };
```
Comment (jarrodruby) [inline server/controller/apiController.js:51]: Let's log the actually error. Just for our debug

**Review by jarrodruby — APPROVED** [review]
Some minor comments

File: server/controller/apiController.js (hunk 1 above)
Comment (karleenmsrichards) [reply server/controller/apiController.js:51]: Hi @jarrodruby , will add the changes to my next pull request, thanks much.

File: server/api.js [hunk 2]
```diff
@@ -48,4 +51,59 @@ router.post("/validation", async (req, res) => {
...
+			profession,
+			yearsOfExperience,
+			hourlyRate,
+			language,
+		});
+
+		res.status(201).json(result);
+	} catch (error) {
+		res.status(500).json({ error: "Internal server error" });
+	}
```
[the remark this answers is not in the document]
Comment (karleenmsrichards) [reply server/api.js:106]: @jarrodruby Oh I see what you mean, sure 😀

File: server/controller/apiController.js [hunk 3]
```diff
@@ -1,6 +1,53 @@
...
+	country,
+	profession,
+	yearsOfExperience,
+	hourlyRate,
+	language,
+}) {
+	try {
+		const existingUser = await Users.findOne({ where: { email } });
+
+		if (existingUser && existingUser.role === "customer") {
```
[the remark this answers is not in the document]
Comment (karleenmsrichards) [reply server/controller/apiController.js:27]: @jarrodruby this is true, I will make the changes. Thank you.

File: server/controller/apiController.js (hunk 1 above)
Comment (karleenmsrichards) [reply server/controller/apiController.js:51]: Hi @Jarrod, for line 50-51, did you mean to log the error or return the actual error.
```console.log(error)``` or ```return { error: error };```

**Pull request merged**